hamocc/mo_powach.F90

```diff
@@ -0,0 +1,519 @@
+! Copyright (C) 2001  Ernst Maier-Reimer, S. Legutke
+! Copyright (C) 2020  K. Assmann, J. Tjiputra, J. Schwinger
+!
+! This file is part of BLOM/iHAMOCC.
+!
+! BLOM is free software: you can redistribute it and/or modify it under the
+! terms of the GNU Lesser General Public License as published by the Free
+! Software Foundation, either version 3 of the License, or (at your option)
+! any later version.
+!
+! BLOM is distributed in the hope that it will be useful, but WITHOUT ANY
+! WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+! FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for
+! more details.
+!
+! You should have received a copy of the GNU Lesser General Public License
+! along with BLOM. If not, see https://www.gnu.org/licenses/.
+
+module mo_powach
+
+  implicit none
+  private
+
+  public :: powach
+
+contains
+
+  subroutine powach(kpie,kpje,kpke,kbnd,prho,omask,psao,lspin)
+
+    !******************************************************************************
+    ! Ernst Maier-Reimer,    *MPI-Met, HH*    10.04.01
+    ! Modified: S.Legutke,   *MPI-MaD, HH*    10.04.01
+    !******************************************************************************
+
+    use mo_control_bgc, only: dtbgc,use_cisonew
+    use mo_param1_bgc,  only: ioxygen,ipowaal,ipowaic,ipowaox,ipowaph,ipowasi,ipown2,ipowno3,&
+                              isilica,isssc12,issso12,issssil, &
+                              issster,ks,ipowc13,ipowc14,isssc13,isssc14,issso13,issso14,safediv
+    use mo_carbch,      only: co3,keqb,ocetra,sedfluxo
+    use mo_chemcon,     only: calcon
+    use mo_param_bgc,   only: rnit,ro2ut,disso_sil,silsat,disso_poc,sed_denit,disso_caco3
+    use mo_sedmnt,      only: porwat,porsol,powtra,produs,prcaca,prorca,&
+                              seddw,sedhpl,sedlay,silpro,pror13,pror14,prca13,prca14
+    use mo_vgrid,       only: kbo,bolay
+    use mo_powadi,      only: powadi
+    use mo_carchm,      only: carchm_solve
+    use mo_dipowa,      only: dipowa
+
+    ! Arguments
+    integer, intent(in) :: kpie                                         ! 1st dimension of model grid.
+    integer, intent(in) :: kpje                                         ! 2nd dimension of model grid.
+    integer, intent(in) :: kpke                                         ! 3rd (vertical) dimension of model grid.
+    integer, intent(in) :: kbnd                                         ! nb of halo grid points
+    real,    intent(in) :: prho(kpie,kpje,kpke)                         ! seawater density [g/cm^3].
+    real,    intent(in) :: omask(kpie,kpje)                             ! salinity [psu].
```

Comment:
comments for `omask` and `psao` mixed up.

Reply:
Done.